packages/ui-kit/src/components/PieChart/PieChart.tsx

```diff
@@ -0,0 +1,326 @@
+    const [propsMismatch, setPropsMismatch] = React.useState(false)
+
+    const idRef = React.useRef(idCounter++)
+    const id = `piechart-${idRef.current}`
+
+    /**
+     * The html node where the chart will render
+     */
+    const canvasRef = React.useRef<HTMLCanvasElement>(null)
+    const chartRef = React.useRef<ChartJS | null>()
+
+    /**
+     * Checks if the component is in `static` or `connected` mode
+     */
+    const isStatic = !query
+
+    /**
+     * Fetches the leaderboard data from the API
+     */
+    const {
+      data: leaderboardData,
+      isLoading: leaderboardIsLoading,
+      error: leaderboardHasError
+    } = useLeaderboard({ ...query })
+
+    /**
+     * Fetches the counter data from the API
+     */
+    const { data: counterData, isLoading: counterIsLoading, error: counterHasError } = useCounter({ ...query })
+
+    const isLoading = leaderboardIsLoading || counterIsLoading
+
+    const hasError = leaderboardHasError || counterHasError
+
+    const isPie = variant === 'pie'
+    const isDoughnut = variant === 'doughnut'
+
+    const renderChart = React.useCallback(
+      (data?: PieChartdData) => {
+        if (!canvasRef.current || !data || !theme || !chartConfig) {
+          return
+        }
+
+        const { chartColorPlatte = defaultChartColorPlatte } = chartProps
+
+        const labels = (data.rows?.map((row) => row.slice(0, row.length - 1)) as PieChartLabels) ?? []
+
+        const values =
+          data.rows?.map((row) => (row[row.length - 1] === null ? null : Number(row[row.length - 1]))) || []
+
+        const customChartLabelsPlugin: Plugin<'pie' | 'doughnut'> = getCustomChartLabelsPlugin({
+          theme
+        })
+
+        const customPlugins = {
+          customCanvasBackgroundColor: {
+            color: card ? theme?.bgPrimary : 'transparent'
+          },
+          legend: {
+            display: true,
+            labels: {
+              usePointStyle: true,
+              pointStyle: '*',
+              pointStyleWidth: 8,
+              boxHeight: 6
+            }
+          },
+          customChartLabelsPlugin
+        }
+
+        const datasets = isDoughnut ? { cutout: '75%' } : { cutout: '0' }
+
+        if (chartRef.current) {
+          const chart = chartRef.current
+
+          chart.data.labels = labels
+          Object.assign(chart.data.datasets[0], {
+            type: variant,
+            data: values,
+            backgroundColor: chartColorPlatte,
+            ...datasets
+          })
+
+          chart.options.plugins = {
+            ...chart.options.plugins,
+            ...customPlugins
+          }
+
+          chart.update()
+          return
+        }
+
+        let config: ChartConfiguration<'pie' | 'doughnut'> = {
+          ...chartConfig,
+          type: variant,
+          data: {
+            labels: labels,
+            datasets: [
+              {
+                data: values as number[],
```

Comment:
Following with my comment above this casting is dangerous as `values` type is not actually `number[]` it could contain `null` values